spec/support/evm_spec_helper.rb

```diff
@@ -47,7 +45,7 @@ def self.clear_caches
     # Clear the thread local variable to prevent test contamination
     User.current_user = nil if defined?(User) && User.respond_to?(:current_user=)
 
-    ::Settings.reload! if @settings_loaded != Vmdb::Settings.last_loaded
+    ::Settings.reload!
```

Comment:
This is unfortunate as it's going to add unnecessary database queries across the entire suite.  Wonder if there's a way we can preserve this locally?

Reply:
So I didn't really understand what this was accomplishing. Is this ensuring that settings are not changed by a spec? How could that happen if we're rolling back the database between specs?

Reply:
It ensures that the top-level settings constant is returned back to the beginning.  Since ::Settings is effectively a cache of the on-disk files and some database records, rollback back is not sufficient to bust this "cache".  So, this line busts the cache, but the way it works it has to query the database (which will return nothing, but the query still happens).  The if clause was a performance hack to prevent this cache bust from happening every time if it wasn't necessary, since in reality 99% of the tests won't mess with settings.

Reply:
we could md5 the Settings json before the `yield` on 33 and check it here:

```ruby
::Settings.reload! if Digest::MD5.hexdigest(::Settings.to_json) != json_md5
```

It's about 5 times faster than reload! everytime (if settings is changed 10% of the time)

```
Comparison:
reload! if json md5 !=:       61.2 i/s
reload! if yml md5 !=:       34.2 i/s - same-ish: difference falls within error
             reload!:       12.9 i/s - 4.74x  slower
```


from: 

```ruby
require 'benchmark/ips'
require './config/environment'
require 'digest/md5'
::Settings
yml_md5 =  Digest::MD5.hexdigest(YAML.dump(::Settings))
json_md5 = Digest::MD5.hexdigest(::Settings.to_json)

Benchmark.ips do |benchmark|
  benchmark.report("reload!") do
    ::Settings.reload!
  end

  benchmark.report("reload! if json md5 !=") do
    if rand(9) == 1
      ::Settings.api.token_ttl = "1.minute"
    end
    ::Settings.reload! if Digest::MD5.hexdigest(::Settings.to_json) != json_md5
  end

  benchmark.report("reload! if yml md5 !=") do
    if rand(9) == 1
      ::Settings.api.token_ttl = "1.minute"
    end
    ::Settings.reload! if Digest::MD5.hexdigest(YAML.dump(::Settings)) != yml_md5
  end

  benchmark.compare!
end
```

Reply:
json seems good...technically there would be minor things like if someone sets a timestamp value to a string value with the exact same value, but that feels very edge case.